pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -150,13 +156,67 @@ func (dv *openShiftClusterDynamicValidator) validateServicePrincipalProfile(ctx
 	return autorest.NewBearerAuthorizer(token), nil
 }
 
-func (dv *openShiftClusterDynamicValidator) validateVnetPermissions(ctx context.Context, oc *api.OpenShiftCluster, client authorization.PermissionsClient, code, typ string) error {
-	vnetID, _, err := subnet.Split(oc.Properties.MasterProfile.SubnetID)
+func (dv *openShiftClusterDynamicValidator) validateVnetPermissions(ctx context.Context, vnet mgmtnetwork.VirtualNetwork, oc *api.OpenShiftCluster, client authorization.PermissionsClient, code, typ string) error {
+	r, err := azure.ParseResourceID(*vnet.ID)
 	if err != nil {
 		return err
 	}
 
-	r, err := azure.ParseResourceID(vnetID)
+	timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+
+	err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
```

Comment:
can all of this implementation be shared rather than copied?

Reply:
I'll give this a try.